From e94ba0619477e990903abd0ea0ab584b98dbac08 Mon Sep 17 00:00:00 2001
From: Dmitry Khlestkov <dsx.box@yandex.ru>
Date: Sat, 24 Feb 2018 07:15:07 -0700
Subject: [PATCH] [UDisks] Fix auto-mount regression

Summary:
BUG: 389479
Devices with empty path stopped auto-mounting after commit 1384f275ab2f1ad1841753ee163af6d1b0bb952b.
This diff fixes the regression by checking if the path is empty.

Reviewers: #frameworks, broulik

Reviewed By: broulik

Subscribers: ngraham, sefaeyeoglu, rikmills, #frameworks

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D10671
---
 src/solid/devices/backends/udisks2/udisksstorageaccess.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/solid/devices/backends/udisks2/udisksstorageaccess.cpp b/src/solid/devices/backends/udisks2/udisksstorageaccess.cpp
index 5630e806cb..dd8f76fa20 100644
--- a/src/solid/devices/backends/udisks2/udisksstorageaccess.cpp
+++ b/src/solid/devices/backends/udisks2/udisksstorageaccess.cpp
@@ -111,7 +111,8 @@ bool StorageAccess::isIgnored() const
 
     const QString path = filePath();
 
-    return !path.startsWith(QLatin1String("/media/"))
+    return !path.isEmpty()
+               && !path.startsWith(QLatin1String("/media/"))
                && !path.startsWith(QLatin1String("/run/media/"))
                && !path.startsWith(QDir::homePath());
 }
